pkg/agent/apiserver/handlers/fqdncache/handler_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2025 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fqdncache
+
+import (
+	"encoding/json"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"antrea.io/antrea/pkg/agent/types"
+	queriertest "antrea.io/antrea/pkg/querier/testing"
+)
+
+func TestFqdnCacheQuery(t *testing.T) {
+	tests := []struct {
+		name             string
+		expectedStatus   int
+		expectedResponse []types.DnsCacheEntry
+	}{
+		{
+			name:           "FQDN cache exists - multiple addresses multiple domains",
+			expectedStatus: http.StatusOK,
+			expectedResponse: []types.DnsCacheEntry{
```

Comment:
Rename this field to `filteredCacheEntries` and make `expectedResponse` a field of type `[]FQDNCacheResponse` instead. It will feel like you are duplicating data, but really it will lead to a cleaner test.